src/commands/add.ts

```diff
@@ -28,26 +29,24 @@ export const builder: { [key: string]: yargs.Options } = {};
 
 export function handler(): Promise<unknown> {
   logInfo(":palm_tree: Getting a break? Great!\n");
-  return askWhere()
-    .chain(where =>
-      askWhen().chain(when => {
+  return mapLeft(
+    taskEither.chain(askWhere(), where =>
+      taskEither.chain(askWhen(), when => {
```

Comment:
😟

- business logic flow is lost
- we begin with `mapLeft`, which is then only "closed" at line 48

Reply:
Have you also experimented with `Do` or `Fluent` for these longer business logic portions?

Reply:
I'm about to